core-web/libs/portlets/dot-content-drive/portlet/src/lib/shared/services/dot-content-drive-navigation.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable, inject } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { take } from 'rxjs/operators';
+
+import { DotContentTypeService } from '@dotcms/data-access';
+import { DotContentDriveItem, FeaturedFlags } from '@dotcms/dotcms-models';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class DotContentDriveNavigationService {
+    readonly #router = inject(Router);
+    readonly #dotContentTypeService = inject(DotContentTypeService);
+
+    /**
+     * Navigates to the appropriate editor based on the content type
+     * @param contentlet The content item to edit
+     */
+    editContent(contentlet: DotContentDriveItem) {
+        if (contentlet.contentType === 'htmlpageasset') {
+            this.#editPage(contentlet);
+        } else {
+            this.#editContentlet(contentlet);
+        }
+    }
+
+    #editPage(contentlet: DotContentDriveItem) {
+        const url = contentlet.urlMap || contentlet.url;
+        this.#router.navigate(['/edit-page/content'], {
+            queryParams: { url, language_id: contentlet.languageId }
+        });
+    }
```

Comment:
Don't we have a global routing service for this?

Reply:
No, we don't have a routing service yet.

Reply:
https://github.com/dotCMS/core/blob/613166823d042c9853bfcfaed83bf0ae864d35e9/core-web/libs/data-access/src/lib/dot-router/dot-router.service.ts#L114-L115

???